pkg/asset/installconfig/platformpermscheck.go

```diff
@@ -58,7 +59,20 @@ func (a *PlatformPermsCheck) Generate(dependencies asset.Parents) error {
 		if err != nil {
 			return errors.Wrap(err, "validate AWS credentials")
 		}
-	case azure.Name, baremetal.Name, gcp.Name, libvirt.Name, none.Name, openstack.Name, ovirt.Name, vsphere.Name:
+	case gcp.Name:
+		client, err := gcpconfig.NewClient(context.TODO())
+		if err != nil {
+			return err
+		}
+		services, err := client.GetEnabledServices(ctx, "projects/"+ic.Config.GCP.ProjectID)
```

Comment:
rather than calling this here in platformpermscheck, I would pass the client to your validation function and get the services there.